### ISSUE TITLE
jekyll 3 beta 9 doesn't generate posts with missing timezone

### ISSUE THREAD

**Issue by marpala** [jekyll/jekyll#4010]
*Jekyll 3.0.0.pre.beta9* freshly installed on El Capitan doesn't seem to process markdown files and it returns no error.

![jekyll3 folder](https://i.imgur.com/45Xkpyc.png)

![jekyll3 terminal](https://i.imgur.com/IdusYmx.png)

![jekyll3 terminal](https://i.imgur.com/81g8DwS.png)

**Comment by jaybe-jekyll** [comment]
Works as expected for me. (10.11; 15A284)

How are you choosing to install jekyll and its dependencies?

My approach:

1. Install Xcode/Development environment.
2. Install jekyll and friends (user-locally)
  1. `gem install --prerelease --no-rdoc --no-ri --bindir ~/bin --no-document --user-install --verbose jekyll`
3. `export PATH=$PATH:~/bin`
4. `jekyll --version && which jekyll`
 
> ```sh
$ jekyll --version && which jekyll
jekyll 3.0.0.pre.beta9
/Users/user1/bin/jekyll
```

5. `jekyll new test` && `cd test`
6. `jekyll serve`

> ```sh
$ jekyll serve
Configuration file: /Users/user1/jekyll/site/_config.yml
            Source: /Users/user1/jekyll/site
       Destination: /Users/user1/jekyll/site/_site
 Incremental build: enabled
      Generating... 
                    done in 0.157 seconds.
 Auto-regeneration: enabled for '/Users/user1/jekyll/site'
Configuration file: /Users/user1/jekyll/site/_config.yml
    Server address: http://127.0.0.1:4000/
  Server running... press ctrl-c to stop.
```

![screen shot 2015-10-05 at 3 21 40 pm](https://cloud.githubusercontent.com/assets/5395940/10292421/cc7e2406-6b74-11e5-98fa-ff28b5a50a11.png)

**Comment by FourFIFTY** [comment]
I installed jekyll with:
`gem install jekyll -pre`

Also, I manage rubies with Rbenv:
`➜  ~   jekyll --version && which jekyll
jekyll 3.0.0.pre.beta9
/Users/Marco/.rbenv/shims/jekyll`

**Comment by envygeeks** [comment]
Did you rehash after?

**Comment by marpala** [comment]
I did.  My complete process was:
*Fresh El Capitan install > Xcode > Homerew > Rbenv + latest ruby (with rehash and correct PATH) > Jekyll pre*
Jekyll is the only ruby-related thing I installed, I didn't touch anything else.

**Comment by envygeeks** [comment]
Can you please kick on verbose mode `--verbose` and post the output, I'm interested to see what is going on, though I don't know if we log that deeply (we should.)

**Comment by marpala** [comment]
Sure

>➜  jekyll-beta9-test  jekyll serve --verbose
  Logging at level: debug
Configuration file: /Users/Marco/Sites/jekyll-beta9-test/_config.yml
            Source: /Users/Marco/Sites/jekyll-beta9-test
       Destination: /Users/Marco/Sites/jekyll-beta9-test/_site
 Incremental build: enabled
      Generating...
       EntryFilter: excluded?(/default.html) ==> false
       EntryFilter: excluded?(/page.html) ==> false
       EntryFilter: excluded?(/post.html) ==> false
       EntryFilter: excluded?(/about.md) ==> false
       EntryFilter: excluded?(/css) ==> false
       EntryFilter: excluded?(/feed.xml) ==> false
       EntryFilter: excluded?(/index.html) ==> false
       EntryFilter: excluded?(/_posts/2015-10-05-welcome-to-jekyll.markdown) ==> false
       EntryFilter: excluded?(/css/main.scss) ==> false
                    done in 0.105 seconds.
           Watcher: Ignoring (?-mix:_config\.yml)
           Watcher: Ignoring (?-mix:_site)
 Auto-regeneration: enabled for '/Users/Marco/Sites/jekyll-beta9-test'
  Logging at level: debug
Configuration file: /Users/Marco/Sites/jekyll-beta9-test/_config.yml
[2015-10-07 11:55:42] INFO  WEBrick 1.3.1
[2015-10-07 11:55:42] INFO  ruby 2.2.3 (2015-08-18) [x86_64-darwin15]
[2015-10-07 11:55:42] DEBUG WEBrick::HTTPServlet::FileHandler is mounted on /.
[2015-10-07 11:55:42] DEBUG unmount .
[2015-10-07 11:55:42] DEBUG #<Class:0x007fd652c5cb98> is mounted on .
    Server address: http://127.0.0.1:4000/
[2015-10-07 11:55:42] INFO  WEBrick::HTTPServer#start: pid=68565 port=4000
  Server running... press ctrl-c to stop.

**Comment by envygeeks** [comment]
You'll need to remove `_site` to get us the full log because with it there incremental regeneration will take over and skip the build.

**Comment by marpala** [comment]
After I deleted `_site`:
➜  jekyll-beta9-test  jekyll serve --verbose
  Logging at level: debug
Configuration file: /Users/Marco/Sites/jekyll-beta9-test/_config.yml
            Source: /Users/Marco/Sites/jekyll-beta9-test
       Destination: /Users/Marco/Sites/jekyll-beta9-test/_site
 Incremental build: enabled
      Generating...
       EntryFilter: excluded?(/default.html) ==> false
       EntryFilter: excluded?(/page.html) ==> false
       EntryFilter: excluded?(/post.html) ==> false
       EntryFilter: excluded?(/about.md) ==> false
       EntryFilter: excluded?(/css) ==> false
       EntryFilter: excluded?(/feed.xml) ==> false
       EntryFilter: excluded?(/index.html) ==> false
       EntryFilter: excluded?(/_posts/2015-10-07-welcome-to-jekyll.markdown) ==> false
       EntryFilter: excluded?(/css/main.scss) ==> false
                    done in 0.179 seconds.
           Watcher: Ignoring (?-mix:_config\.yml)
           Watcher: Ignoring (?-mix:_site)
 Auto-regeneration: enabled for '/Users/Marco/Sites/jekyll-beta9-test'
  Logging at level: debug
Configuration file: /Users/Marco/Sites/jekyll-beta9-test/_config.yml
[2015-10-07 12:02:17] INFO  WEBrick 1.3.1
[2015-10-07 12:02:17] INFO  ruby 2.2.3 (2015-08-18) [x86_64-darwin15]
[2015-10-07 12:02:17] DEBUG WEBrick::HTTPServlet::FileHandler is mounted on /.
[2015-10-07 12:02:17] DEBUG unmount .
[2015-10-07 12:02:17] DEBUG #<Class:0x007ff12134a228> is mounted on .
    Server address: http://127.0.0.1:4000/
[2015-10-07 12:02:17] INFO  WEBrick::HTTPServer#start: pid=68803 port=4000
  Server running... press ctrl-c to stop.

**Comment by envygeeks** [comment]
@parkr does the logger not log generations?

**Comment by parkr** [comment]
Nope :frowning:

Did you try with --full-rebuild?

**Comment by marpala** [comment]
`--full-rebuild` worked but it didn't process new files in _posts, only the first one. Moreover, it didn't work at all on a newly generated blog.

**Comment by parkr** [comment]
> it didn't work at all on a newly generated blog.

What do you mean by this? When you ran `jekyll build --full-rebuild` on a brand new site, it didn't work?

**Comment by marpala** [comment]
No sorry it did work, but it didn't generate post files. For some reason, running the command with `--full-rebuild` generated post files only once on that first site I showed you in the screenshots and then stopped generating both on that site and on all the other sites I created later.
I should add that jekyll generates pages for markdown files in the root directory with no problem.

**Comment by marpala** [comment]
Quick update. Yesterday I deleted all the sites I had created and then created a new one called "test". None of the building commands (serve, build, build --full-rebuild) generated post files for that site.
Then today I gave it another try and for some reason it finally worked. So I decided to test it out on a new project, but it didn't generate anything again.

It's very random.

**Comment by envygeeks** [comment]
I don't know, that to me sounds like a systems issue, I would think an issue like this wouldn't be so intermittent with a persistent configuration.  Since I'm a lazy guy today, can you fill me in on the env so I can try and replicate it sometime this week.

**Comment by marpala** [comment]
Sure
```
➜  ~  gem env
RubyGems Environment:
  - RUBYGEMS VERSION: 2.4.8
  - RUBY VERSION: 2.2.3 (2015-08-18 patchlevel 173) [x86_64-darwin15]
  - INSTALLATION DIRECTORY: /Users/Marco/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0
  - RUBY EXECUTABLE: /Users/Marco/.rbenv/versions/2.2.3/bin/ruby
  - EXECUTABLE DIRECTORY: /Users/Marco/.rbenv/versions/2.2.3/bin
  - SPEC CACHE DIRECTORY: /Users/Marco/.gem/specs
  - SYSTEM CONFIGURATION DIRECTORY: /Users/Marco/.rbenv/versions/2.2.3/etc
  - RUBYGEMS PLATFORMS:
    - ruby
    - x86_64-darwin-15
  - GEM PATHS:
     - /Users/Marco/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0
     - /Users/Marco/.gem/ruby/2.2.0
  - GEM CONFIGURATION:
     - :update_sources => true
     - :verbose => true
     - :backtrace => false
     - :bulk_threshold => 1000
  - REMOTE SOURCES:
     - https://rubygems.org/
  - SHELL PATH:
     - /Users/Marco/.rbenv/versions/2.2.3/bin
     - /usr/local/Cellar/rbenv/0.4.0/libexec
     - /Users/Marco/.rbenv/shims
     - /Users/Marco/.rbenv/bin
     - /usr/local/bin
     - /usr/bin
     - /bin
     - /usr/sbin
     - /sbin
```

**Comment by parkr** [comment]
@marpala If you can repro on GitHub Pages or in a Travis build, that would help us. El Capitan is still pretty gross for developers so I don't want to upgrade quite yet. Thanks for being the guinea pig here.

Can you clone down this repo, then run `script/bootstrap` and `script/cibuild`?

**Comment by marpala** [comment]
Alright, I cloned the repo and ran those commands. What should I put on Github Pages?

**Comment by parkr** [comment]
@marpala Did you get any errors from script/cibuild? This issue seems like a wontfix because it happens intermittently and cannot be consistently reproduced by others.

**Comment by marpala** [comment]
Nope I didn't

**Comment by envygeeks** [comment]
I am also unable to produce it on any of my machines (Linux/Mac) -- I am also not able to re-produce it in Docker.  I just won't know how it's caused but I'm falling back on what I said when I chimed in... this could be a systems issue.  If you can replicate it please do give steps on how we can so we can debug it (I'll be the first to jump on it since this is an intriguing problem,) however for now I agree with @parkr, it's probably a WONTFIX or CANTFIX because we simply cannot fix something we cannot replicate, maybe @mattr- or @alfredxing want to take a stab at reproducing but I am absolutely unable to.

**Comment by parkr** [comment]
Going to close as WONTFIX for now. If others have this issue, they can chime in and we can revisit it. We see problems like this once or twice a year and it's generally system issues with Listen or permissions issues. Thanks!

**Comment by meskyanichi** [comment]
I too have this issue. Installed the latest Jekyll under the latest Ruby (using ruby-install/chruby). When generating a new website and running `jekyll serve` I see the home page without the initial post listed (exactly as in OP's screenshot).

Two things I've noticed:

- I need to use `jekyll serve --full-rebuild`, else the changes won't be compiled.
- The original post that's generated in `_posts` uses my local time (Amsterdam) in the front matter. When I subtract 2 hours from this time, the post does display on the home page. Though, again, only when using the `--full-rebuild` flag. Without `--full-rebuild` it doesn't appear regardless.

Also running a fresh install of El Capitan.

```
which jekyll && jekyll --version && which ruby && ruby --version
/Users/Michael/.gem/ruby/2.2.3/bin/jekyll
jekyll 3.0.0.pre.beta9
/Users/Michael/.rubies/ruby-2.2.3/bin/ruby
ruby 2.2.3p173 (2015-08-18 revision 51636) [x86_64-darwin15]
```

The process:

```
# generate new website
Desktop $ jekyll new example
New jekyll site installed in /Users/Michael/Desktop/example.

# move into website
Desktop $ cd example

# boot server in verbose mode
example $ jekyll serve --verbose
  Logging at level: debug
Configuration file: /Users/Michael/Desktop/example/_config.yml
            Source: /Users/Michael/Desktop/example
       Destination: /Users/Michael/Desktop/example/_site
 Incremental build: enabled
      Generating...
       EntryFilter: excluded?(/default.html) ==> false
       EntryFilter: excluded?(/page.html) ==> false
       EntryFilter: excluded?(/post.html) ==> false
       EntryFilter: excluded?(/about.md) ==> false
       EntryFilter: excluded?(/css) ==> false
       EntryFilter: excluded?(/feed.xml) ==> false
       EntryFilter: excluded?(/index.html) ==> false
       EntryFilter: excluded?(/_posts/2015-10-13-welcome-to-jekyll.markdown) ==> false
       EntryFilter: excluded?(/css/main.scss) ==> false
                    done in 0.199 seconds.
           Watcher: Ignoring (?-mix:_config\.yml)
           Watcher: Ignoring (?-mix:_site)
 Auto-regeneration: enabled for '/Users/Michael/Desktop/example'
  Logging at level: debug
Configuration file: /Users/Michael/Desktop/example/_config.yml
[2015-10-13 18:03:35] INFO  WEBrick 1.3.1
[2015-10-13 18:03:35] INFO  ruby 2.2.3 (2015-08-18) [x86_64-darwin15]
[2015-10-13 18:03:35] DEBUG WEBrick::HTTPServlet::FileHandler is mounted on /.
[2015-10-13 18:03:35] DEBUG unmount .
[2015-10-13 18:03:35] DEBUG #<Class:0x007fd649bc2f10> is mounted on .
    Server address: http://127.0.0.1:4000/
[2015-10-13 18:03:35] INFO  WEBrick::HTTPServer#start: pid=2010 port=4000
  Server running... press ctrl-c to stop.

# GET / -- no posts listed on home page
[2015-10-13 18:03:47] DEBUG accept: 127.0.0.1:51330
[2015-10-13 18:03:47] DEBUG  is invoked.
127.0.0.1 - - [13/Oct/2015:18:03:47 CEST] "GET / HTTP/1.1" 200 5161
- -> /
[2015-10-13 18:03:47] DEBUG  is invoked.
127.0.0.1 - - [13/Oct/2015:18:03:47 CEST] "GET /css/main.css HTTP/1.1" 200 8732
http://localhost:4000/ -> /css/main.css
[2015-10-13 18:03:47] DEBUG accept: 127.0.0.1:51332
[2015-10-13 18:03:47] DEBUG  is invoked.
[2015-10-13 18:03:47] ERROR `/favicon.ico' not found.
127.0.0.1 - - [13/Oct/2015:18:03:47 CEST] "GET /favicon.ico HTTP/1.1" 404 281
http://localhost:4000/ -> /favicon.ico

# Make a change to `_posts/2015-10-13-welcome-to-jekyll.markdown` (subtract 2 hours from `date` front matter)
[2015-10-13 18:03:47] DEBUG close: 127.0.0.1:51332
      Regenerating: 1 file(s) changed at 2015-10-13 18:03:59        EntryFilter: excluded?(/default.html) ==> false
       EntryFilter: excluded?(/page.html) ==> false
       EntryFilter: excluded?(/post.html) ==> false
       EntryFilter: excluded?(/about.md) ==> false
       EntryFilter: excluded?(/css) ==> false
       EntryFilter: excluded?(/feed.xml) ==> false
       EntryFilter: excluded?(/index.html) ==> false
       EntryFilter: excluded?(/_posts/2015-10-13-welcome-to-jekyll.markdown) ==> false
       EntryFilter: excluded?(/css/main.scss) ==> false
...done in 0.250212 seconds.

# Shutdown server and restart with --full-rebuild
example $ jekyll serve --full-rebuild --verbose
  Logging at level: debug
Configuration file: /Users/Michael/Desktop/example/_config.yml
            Source: /Users/Michael/Desktop/example
       Destination: /Users/Michael/Desktop/example/_site
 Incremental build: disabled
      Generating...
       EntryFilter: excluded?(/default.html) ==> false
       EntryFilter: excluded?(/page.html) ==> false
       EntryFilter: excluded?(/post.html) ==> false
       EntryFilter: excluded?(/about.md) ==> false
       EntryFilter: excluded?(/css) ==> false
       EntryFilter: excluded?(/feed.xml) ==> false
       EntryFilter: excluded?(/index.html) ==> false
       EntryFilter: excluded?(/_posts/2015-10-13-welcome-to-jekyll.markdown) ==> false
       EntryFilter: excluded?(/css/main.scss) ==> false
                    done in 0.281 seconds.
           Watcher: Ignoring (?-mix:_config\.yml)
           Watcher: Ignoring (?-mix:_site)
 Auto-regeneration: enabled for '/Users/Michael/Desktop/example'
  Logging at level: debug
Configuration file: /Users/Michael/Desktop/example/_config.yml
[2015-10-13 18:10:18] INFO  WEBrick 1.3.1
[2015-10-13 18:10:18] INFO  ruby 2.2.3 (2015-08-18) [x86_64-darwin15]
[2015-10-13 18:10:18] DEBUG WEBrick::HTTPServlet::FileHandler is mounted on /.
[2015-10-13 18:10:18] DEBUG unmount .
[2015-10-13 18:10:18] DEBUG #<Class:0x007f8f9bad9850> is mounted on .
    Server address: http://127.0.0.1:4000/
[2015-10-13 18:10:18] INFO  WEBrick::HTTPServer#start: pid=2084 port=4000
  Server running... press ctrl-c to stop.

# Get / -- now I see the post on the home page  
[2015-10-13 18:10:32] DEBUG accept: 127.0.0.1:51377
[2015-10-13 18:10:32] DEBUG  is invoked.
127.0.0.1 - - [13/Oct/2015:18:10:32 CEST] "GET / HTTP/1.1" 200 5379
- -> /
[2015-10-13 18:10:32] DEBUG  is invoked.
127.0.0.1 - - [13/Oct/2015:18:10:32 CEST] "GET /css/main.css HTTP/1.1" 200 8732
http://localhost:4000/ -> /css/main.css
```

**Comment by parkr** [comment]
> The original post that's generated in _posts uses my local time (Amsterdam) in the front matter. When I subtract 2 hours from this time, the post does display on the home page.

!

So perhaps it's a timing issue with the `date` attribute? What does that look like in your post?

**Comment by meskyanichi** [comment]
Initially, in the front matter of `./_posts/2015-10-13-welcome-to-jekyll.markdown`:


```yaml
---
layout: post
title:  "Welcome to Jekyll!"
date:   2015-10-13 19:29:04
categories: jekyll update
---
```

No posts are shown on the home page. Updating the front matter to:

```yaml
---
layout: post
title:  "Welcome to Jekyll!"
date:   2015-10-13 17:29:04
categories: jekyll update
---
```

Makes the post appear on the home page (when using `--full-rebuild`).

**Comment by envygeeks** [comment]
Ruby bug on new OS X? We haven't adjusted any of that lately as far as I remember, Occam is telling me to say it could be Ruby but only once it's confirmed we've not adjusted anything in that area, but seriously I don't recall anything being adjusted there?

**Comment by alfredxing** [comment]
I couldn't repro on my machine (I'm running El Cap as well). My Ruby is installed with Homebrew (`ruby 2.2.3p173 (2015-08-18 revision 51636) [x86_64-darwin15]`).

What I did:

1. Install Jekyll 3.0.0.pre.beta9 using `gem install jekyll --pre`
2. Create a new test site: `jekyll _3.0.0.pre.beta9_ new test` (note here that using `_3.0.0.pre.beta9_` as the first argument specifies the version of the gem to use; this seems to be built in to the `jekyll` script generated by RubyGems. This _shouldn't_ be required, but I needed it since I had regular 2.5.3 Jekyll installed as well)
3. `cd` into the directory and run `jekyll _3.0.0.pre.beta9_ serve`
4. Go to `http://localhost:4000` in browser, verify that the "Welcome to Jekyll" post is listed.

**Comment by marpala** [comment]
@meskyanichi:
>The original post that's generated in _posts uses my local time (Amsterdam) in the front matter. When I subtract 2 hours from this time, the post does display on the home page. Though, again, only when using the --full-rebuild flag. Without --full-rebuild it doesn't appear regardless.

That would explain why for me it didn't generate posts right away on a newly created site, but only after several hours. Then it stopped working again on new posts (with the current local time in the front matter). My local time is GMT+2 as well.

**Comment by parkr** [comment]
> it only processes posts using the --full-rebuild flag.

That sounds like an incremental regeneration bug. @alfredxing?, know what's happening here?

The new site should generate a post whose date has the time zone offset. Unclear why that isn't in there anymore.

**Comment by alfredxing** [comment]
Not sure, since I can't reproduce the issue. Behind the scenes, incremental regen just checks the file modification time to see if it has changed. Maybe something to do with Ruby dates getting messed up there as well?

**Comment by tobyxnet** [comment]
I'm experiencing the exact same thing on a fresh 10.11 (15A284) El Capitan with current Xcode and beta 9. I installed Jekyll with `gem install jekyll --pre` and created a new scaffolding with `gem new lala`. There is no older jekyll installed.

If I can contribute to the bug hunt somehow, let me know. I apologize profusely if my information is insufficient.

**Comment by parkr** [comment]
There was a timezone bug that I ran into only after I started living in Berlin that perplexed me for a few days. I fixed it, visited Boston, ran my tests again while in the US and they were failing. Time is hard.

Thinking this is related to #4001.

What Ruby versions are y'all using, by the way?

Please try this new beta version, installed the same way you got beta9: https://github.com/jekyll/jekyll/releases/tag/v3.0.0.pre.beta10

**Comment by meskyanichi** [comment]
I've just upgraded to Jekyll `3.0.0.pre.beta10` and generated a new project. The posts now show up immediately without any `date` front matter changes, and the date is generated with the +0200 tz suffix. Additionally, incremental builds appear to be working as well now (from Amsterdam, GMT+2).

Not sure if the following is intentional or not, but I thought I'd point it out:

When I edit the front matter or content of the initial post (or any other post presumably), then the `index.html` that contains the list of posts doesn't get rebuilt. When I save the `index.html` (either with or without changes) it'll rebuild the `index.html` and it'll contain the changes that were made in the initial post (such as a `title` or `date` change).

The actual post that you're editing does get rebuilt as expected. It's just the `index.html` that lists these posts that doesn't pick up the changes.

**Comment by marpala** [comment]
Beta 10 is also finally processing posts for me. (Ruby 2.2.3).

I don't know if this is expected behaviour, but when I add a markdown file in the _posts folder while in watch mode, jekyll returns:
> Regenerating: 1 file(s) changed at 2015-10-17 12:51:25 ...done in 0.104838 seconds.
but in reality nothing shows up in _site until I quit and restart `jekyll serve` with `--full-rebuild`.

**Comment by envygeeks** [comment]
@marpala I have a feeling it's because a TZ was added to the default post which makes the surface seem clean but we don't add a default TZ anywhere else other than setting ENV["TZ"] to truly fix this bug we should add a TZ to all times, I'd have to go back and double check (or somebody else) but that would result in this edge from experience.

**Comment by tobyxnet** [comment]
@parkr beta10 seems to run fine for now. Creating a new blog with `jekyll new thingie` and generating a site from that doesn't leave out the posts anymore.

My Ruby version is ruby 2.0.0p645 (2015-04-13 revision 50299) [universal.x86_64-darwin15].

I'm in CEST, too (Berlin, UTC+2, currently DST adjusted).

**Comment by parkr** [comment]
> When I edit the front matter or content of the initial post (or any other post presumably), then the index.html that contains the list of posts doesn't get rebuilt.

@alfredxing Are dependencies like this tracked in incremental regen?

**Comment by alfredxing** [comment]
@parkr Unfortunately, that's one of the only things incremental regen doesn't track&mdash;dependencies based on looping though posts/pages. The solution at this point is to add `regenerate: true` to the front-matter of `index.html`, which we probably should be doing...

**Comment by meskyanichi** [comment]
I now have an `articles.html` and oddly enough this **does** automatically regenerate whenever a post's front matter is updated (great). I'm not using the `regenerate: true` option here either. It's just the `index.html` that doesn't regenerate unless you specify `regenerate: true` from what I can see.

```html
---
layout: default
permalink: /articles/
title: Articles
---

{% include header.html %}

<div id="articles">
  <h1>Articles</h1>

  <div>
    {% for post in site.posts %}
      <a
        class="article"
        href="{{ post.url | prepend: site.baseurl }}">

        <div class="title">
          {{ post.title }}
        </div>

        <span class="date">
          {{ post.date | date: "%B %-d, %Y" }}
        </span>
      </a>
    {% endfor %}
  </div>
</div>
```

**Comment by envygeeks** [comment]
That's part of the broader issue: https://github.com/jekyll/jekyll/issues/3747

**Comment by yratof** [comment]
Sma eissue occuring for me now, just updated to jekyll 3 and auto-gen of posts no longer happens

**Comment by marpala** [comment]
@yratof did you add a timezone to your posts? The latest version of jekyll 3 has fixed all the problems for me as long as there's a "+0100" next to the date in my post file.

**Comment by yratof** [comment]
That worked – Fantastic. This should be something that could be documented for future upgraders

Cheers @marco

**Comment by kaaveland** [comment]
We were struggling with a similar issue. Tracked the problem down to the safe_yaml gem: https://github.com/dtao/safe_yaml/blob/master/lib/safe_yaml/parse/date.rb#L25-L27

What this appears to do is to read dates in UTC, then convert them to the local time zone. We are +1, so what happens is that every date that gets parsed by safe_yaml has 1 hour added to it. That means that they are "in the future" and jekyll doesn't want to render them.

We're adding 'Europe/Oslo' to all our posts in order to get correct times on our posts. Might be prudent to get this patched upstream or to not allow safe_yaml to parse these files. As far as I can tell, the way jekylls does this in `utils.rb` behaves correctly (at least in all the timezones I tested it in).

**Comment by yratof** [comment]
I've been using this to created posts with the time + 0100 added to the end
of the generation

https://github.com/yratof/yratof.github.io/blob/master/post.sh

On 6 November 2015 at 12:58, Robin Kåveland Hansen <notifications@github.com

**Comment by yazinsai** [comment]
+1, thanks for sharing!

**Comment by kleinfreund** [comment]
What's the status on this? This is still around and quite annoying to track down. Things behave exactly as described by @kaaveland on my end.

**Comment by parkr** [comment]
@kleinfreund Unfortunately I'm not sure what to do here. A first step would be to use UNIX timestamps instead of `Time` instances everywhere. If we can use UNIX for UTC then it's much easier to compare and timezones become merely a presentation detail.

**Comment by kaaveland** [comment]
The workaround of putting the timezone with the timestamp is fine for now, but probably should be documented somewhere to help people who encounter this issue in the future.

The other options would be to file an issue with safe_yaml that this behaviour should be changed (which I would suspect they'd be reluctant to do seeing as there's probably plenty of software that relies on this behaviour at this point) or toggled off somehow.

The path of least resistance would probably be to make everything timezone-aware or to calculate the result from safe_yaml back into utc or unix timestamps. Feels like it would be easy to get this wrong either way, with the dst-minefield and everything here. I'm seeing people add `+0100` etc to their timestamps (which is a timezone without dst), going to guess some people will run into problems with that sort of approach further down the line as well.

Going to sign off with this one: https://www.youtube.com/watch?v=-5wpm-gesOY

**Comment by envygeeks** [comment]
Not to be offensive but if you don't write your times with a zone you're doing time wrong anyways.  How are we supposed to guess what time zone you are in if reading your source? Reading it on Github, reading it anywhere.

**Comment by kaaveland** [comment]
Fair enough, but the actual issue (at least the one we observed) comes up as "I wrote down the current time and jekyll doesn't show my blog post" or some variation of that. I can't quite recall how I ended up in this rabbithole to begin with. Maybe the unreliability of a timestamp without a timezone should be documented somewhere very visible or maybe it should produce a warning? I'm sorry if this is already the case, I didn't see it at the time this was relevant for me.

For what it's worth, I completely agree that guessing at timezones isn't really viable. Currently though, safe_yaml does that for you (it picks the users local timezone). That means that if I send my source without a timezone in my timestamp to a collaborator in India, he won't be able to render it for 3 hours and a half unless he changes the timestamp.

**Comment by envygeeks** [comment]
This is a long standing issue with Ruby itself which never truly handled Time correctly anyways and seems to go out of it's way to mess it up or make it hard every release at one point Jekylll did go out of it's way to try and help alleviate that but it was fragile anyways, over the years it's gotten easier kind-of but it's still broken.  Take a look at Rails; they literally goes out of their way to make this easy by shimming the crap out of `Time`, `DateTime` and `Date` (which that list in-and-of itself should tell you how bad time is in Ruby.)

The problem here if I remember right (I haven't re-read anything again in this ticket) is that we do not infer the timezone I'd have to go back over it, but ultimately let me say this, there are plans in my personal backlog to one day solve problems with time in Jekyll, it's just not an important issue because the lot of us include a TimeZone anyways.

**Comment by djacquel** [comment]
Setting the `timezone` variable in _config.yml can be a solution.

With no `timezone` set :

A date with no offset is considered to be GMT (+0000).

    date : 2015-12-26 00:00:00
    = 2015-12-26 00:00:00 +0000 
    = 2015-12-26 01:00:00 +0100

If you build your site at `2015-12-26 00:10:00 +0100`, your post is in the future and not published, except if you set `future: true` in config or use `jekyll build --future`.

If you set the timezone (eg: `timezone: Europe/Paris` offset +0100 ), any post whith no offset is considered to be +0100.

    date : 2015-12-26 00:00:00 
    = 2015-12-26 00:00:00 +0100

If you build your site at 2015-12-26 00:00:01 +0100, your post is in the past and published.

**Comment by s2t2** [comment]
so thankful for this issue and all the comments.

some of my posts were dated either today or in the future. these showed-up on localhost but not on gh-pages. After changing the post dates to be safely in the past, all the pages now show up.

**Comment by jekyllbot** [comment]
This issue has been automatically marked as stale because it has not been commented on for at least
three months.

The resources of the Jekyll team are limited, and so we are asking for your help.

If you can still reproduce this error on the <code>3.1-stable</code> or <code>master</code> branch,
please reply with all of the information you have about it in order to keep the issue open.

If this is a feature request, please consider building it first as a plugin. Jekyll 3 introduced
[hooks](http://jekyllrb.com/docs/plugins/#hooks) which provide convenient access points throughout
the Jekyll build pipeline whereby most needs can be fulfilled. If this is something that cannot be
built as a plugin, then please provide more information about why in order to keep this issue open.

Thank you for all your contributions.
